pmlc/conversion/gpu_to_spirv/gpu_to_spirv.cc

```diff
@@ -38,6 +38,90 @@ struct StdxSubgroupBroadcastOpConversion
   }
 };
 
+template <typename StdxOpTy, typename SpirvOpTy>
+struct StdxUnaryOpConversion : public SPIRVOpLowering<StdxOpTy> {
+  using SPIRVOpLowering<StdxOpTy>::SPIRVOpLowering;
+
+  LogicalResult
+  matchAndRewrite(StdxOpTy op, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const final {
+    assert(operands.size() == 1);
+    auto dstType = op.getResult().getType();
+    rewriter.replaceOpWithNewOp<SpirvOpTy>(op, dstType, operands.front());
+    return success();
+  }
+};
+
+struct StdxRoundOpConversion : public SPIRVOpLowering<stdx::RoundOp> {
+  using SPIRVOpLowering<stdx::RoundOp>::SPIRVOpLowering;
+
+  LogicalResult
+  matchAndRewrite(stdx::RoundOp op, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const final {
+    assert(operands.size() == 1);
+    auto stdxType = op.getResult().getType();
+    auto loc = op.getLoc();
+    auto cstOp = rewriter.create<spirv::ConstantOp>(
+        loc, stdxType, FloatAttr::get(stdxType, 0.5));
+    auto addOp = rewriter.create<spirv::FAddOp>(loc, cstOp.getResult(),
+                                                operands.front());
+    rewriter.replaceOpWithNewOp<spirv::GLSLFloorOp>(op, stdxType,
+                                                    addOp.getResult());
+
+    return success();
+  }
+};
+
+struct StdxCosHOpConversion : public SPIRVOpLowering<stdx::CosHOp> {
+  using SPIRVOpLowering<stdx::CosHOp>::SPIRVOpLowering;
+  LogicalResult
+  matchAndRewrite(stdx::CosHOp op, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const final {
+    assert(operands.size() == 1);
+    auto loc = op.getLoc();
+    auto operand = operands.front();
+    auto srcType = operand.getType();
+    auto positive_exp =
+        rewriter.create<spirv::GLSLExpOp>(loc, srcType, operand);
+    auto neg_operand = rewriter.create<spirv::FNegateOp>(loc, operand);
+    auto negative_exp =
+        rewriter.create<spirv::GLSLExpOp>(loc, srcType, neg_operand);
+    auto exp_sum =
+        rewriter.create<spirv::FAddOp>(loc, positive_exp, negative_exp);
+    auto cst_2 = rewriter.create<spirv::ConstantOp>(
+        loc, srcType, FloatAttr::get(srcType, 2.0));
+    auto cosh = rewriter.create<spirv::FDivOp>(loc, exp_sum, cst_2);
```

Comment:
Please rename cosh to something else, generally we should avoid using the function names in ordinary headers like math.h as our variable.